packages/odyssey-react-mui/src/PasswordField.tsx

```diff
@@ -59,6 +59,10 @@ export type PasswordFieldProps = {
    * It prevents the user from changing the value of the field
    */
   isReadOnly?: boolean;
+  /**
+   * If `true`, the show/hide icon is not shown to the user
+   */
+  isShowPasswordIconDisabled?: boolean;
```

Comment:
We should make this positive, not negative.

But there's another issue, I typically make default booleans `false`. Is there another way we could name this or create it so it's similar to how we did `isOptional` instead of `isRequired={false}`?

Keep in mind, booleans like `isDisabled` are negative and difficult to wrap your head around. We left that one as it's part of HTML and not the default behavior, but in general, we should only name and use booleans that result in adding a feature rather than removing one.

So in this case, I'm not sure what we can do off the top of my head, but it's a good discussion item. Feel free to start a Slack thread.

Reply:
Another option is using a finite state machine with 2 possible string values.

In TypeScript terms, it'd be called an enum, but at this point, I say we should only use string unions. I might go back to enums in the future, but string unions are my current choice.

Could be passwordIconVisiblity="visible" or "invisible".

---

Since password fields are rarely used, we could also force users to add a prop to their components to be explicit about their intent. That way, any dev that looks at that component could tell exactly how it's gonna render.

Reply:
Would `isPasswordIconHidden` be negative wording? I do agree that its hard to avoid negative wording and make the default boolean `false`. I think using a string is fine, but how do we specify that it has to be those two strings? also, does this mean we set the default value to "visible" or have no default? if there is a default, why is that any better than allowing default values to be true

Reply:
> Since password fields are rarely used, we could also force users to add a prop to their components to be explicit about their intent. That way, any dev that looks at that component could tell exactly how it's gonna render.

my 2c although kind of a breaking change since it _requires_ a new prop/option to be passed

```ts
type PasswordFieldProps: {
  // ... other props
  showPasswordIcon: boolean; // required
}
```